src/DashboardUI/src/contexts/ConservationApplicationState.tsx

```diff
@@ -482,13 +642,64 @@ const onApplicantConsumptiveUseEstimated = (
 
     polygon.fieldName = `Field ${i + 1}`;
     polygon.waterConservationApplicationEstimateLocationId =
-      matchingConsumptiveUseData.waterConservationApplicationEstimateLocationId;
-    polygon.averageYearlyEtInInches = matchingConsumptiveUseData.averageYearlyTotalEtInInches;
-    polygon.averageYearlyEtInAcreFeet = matchingConsumptiveUseData.averageYearlyTotalEtInAcreFeet;
+      matchingConsumptiveUseData.waterConservationApplicationEstimateLocationId ?? undefined;
+    polygon.averageYearlyTotalEtInInches = matchingConsumptiveUseData.averageYearlyTotalEtInInches;
+    polygon.averageYearlyTotalEtInAcreFeet = matchingConsumptiveUseData.averageYearlyTotalEtInAcreFeet;
     polygon.datapoints = matchingConsumptiveUseData.datapoints;
   }
 
   checkCanContinueToApplication(draftState);
+  updatePolygonAcreageSum(draftState);
+  computeCombinedPolygonData(draftState);
+
+  return draftState;
+};
+
+const onReviewerConsumptiveUseEstimateMutationStatusUpdated = (
+  draftState: ConservationApplicationState,
+  { payload }: ReviewerConsumptiveUseEstimateMutationStatusUpdatedAction,
+): ConservationApplicationState => {
+  draftState.reviewerEstimateConsumptiveUseMutationStatus = {
+    ...payload,
+  };
+  return draftState;
+};
+
+const onReviewerConsumptiveUseEstimated = (
+  draftState: ConservationApplicationState,
+  { payload }: ReviewerConsumptiveUseEstimatedAction,
+): ConservationApplicationState => {
+  const application = draftState.conservationApplication;
+
+  application.cumulativeTotalEtInAcreFeet = payload.cumulativeTotalEtInAcreFeet;
+  application.cumulativeNetEtInAcreFeet = payload.cumulativeNetEtInAcreFeet;
+  application.conservationPayment = payload.conservationPayment;
+
+  // combine polygon data
+  for (let i = 0; i < application.estimateLocations.length; i++) {
+    const polygon = application.estimateLocations[i];
+    const matchingConsumptiveUseData = payload.dataCollections.find((data) => data.polygonWkt === polygon.polygonWkt)!;
+
+    application.estimateLocations[i] = {
+      ...application.estimateLocations[i],
+      ...matchingConsumptiveUseData,
+      // preserve Id if possible
+      waterConservationApplicationEstimateLocationId:
+        matchingConsumptiveUseData.waterConservationApplicationEstimateLocationId ??
+        application.estimateLocations[i].waterConservationApplicationEstimateLocationId,
+    };
```

Comment:
```suggestion
    application.estimateLocations[i].waterConservationApplicationEstimateLocationId = matchingConsumptiveUseData.waterConservationApplicationEstimateLocationId ??
        application.estimateLocations[i].waterConservationApplicationEstimateLocationId?
```
is spread needed or can we set the field we want directly?

Reply:
We're not just setting _only_ that field - we're combining the ET data retrieved from the backend into the entry already in state. I'll add a comment here to make the intention more obvious.

---

Now that you point this out though, I think this is code smell for an issue with data preservation.

The intention here is basically that we're doing `Object.assign(application.estimateLocations[i], matchingConsumptiveUseData)` (aka we're taking the matching consumptive use data and we're merging it into the estimate location).

However, assuming that I wrote this code for _some_ reason and didn't just hallucinate it into existence, then there may be a potential issue.

The issue would be that there is a case where the data in state _does_ have an id (for example, if it was retrieved via the `GetApplication` call chain), but the `EstimateConsumptiveUse` response does _not_ have an id for this entry.

Assuming that that issue does exist, then either:
* we never gave the backend the Id for this entry when we asked it to estimate the consumptive use, OR
* the backend did receive the Id for this entry, but it lost that Id at some point along the way (presumably due to an issue with our mapping)

I'll log a bug to investigate this potential issue.

Reply:
bug logged [here](https://dontpaniclabs.visualstudio.com/Western%20States/_workitems/edit/27749)